### ISSUE TITLE
Stuck on Generating RSA Key

### ISSUE THREAD

**Issue by dangoslen** [GoogleCloudPlatform/cloud-sql-jdbc-socket-factory#148]
Hello.

I'm using Google Cloud MySQL + Spring Boot in the Flex GAE environment. I've added the mysql driver and this project for Java8. When I go to make the first connection, the connection gets "stuck" on the line `Generating RSA KeyPair`.

I've looked at some of the other issues around the first connection, but they don't seem to mention getting stuck on this step. 

Any advice? Happy to supply more configuration if you need for troubleshooting.

**Comment by kurtisvg** [comment]
Can you clarify what you mean by getting "stuck"? Does the connection time out? Is it "stuck" indefinitely? 

What version of the driver/socketfactory are you using? Are you using Hikari or some other connection pool?

**Comment by dangoslen** [comment]
Eventually my requests to my service get a 499 - presumably from nginx giving up and closing the connection. When watching the log lines, they simply say "connection to <INSTANCE>" and then the next log says "First connection, generating RSA Key Pair" and then nothing else.

I'm using MySQL Connector 8.0.16 + `mysql-socket-factory-connector-j-8:1.0.14` of this library based on the README. I am using Hikkari as the data source and I'm simply passing all the params in the `url`.

**Comment by kurtisvg** [comment]
It sounds like the `connect` method on the SocketFactory isn't being called. If it was, [this line](    logger.info() would be logged. This seems to indicate it might be a problem somewhere else in your application. 

Can you provide a snippet of how you are attempting to connect?

**Comment by dangoslen** [comment]
Its a Spring Boot Application, just just using the `spring.datasource` properties to init the Hikari `DataSource` and then grabs the `JdbcTemplate` configured from it.

I'll see if I can add some logging, but its good to know that if the `connect` method was getting called then it would proceed.

**Comment by dangoslen** [comment]
So... while I don't have a full explanation for this, as I was debugging further and further, I noticed that I was getting a hanging app on `CloudSqlInstance.createSslSocket()` method. When I looked even deeper, I found that there was a future that was never coming back in the `CloudSqlInstance.getInstanceData()` method.

I narrowed it finally down to `Uninterruptibles.getUninterruptibly`, where the `ListenableFuture` never came back. This utility is part of guava and it looks like I had a particularly old version being declared in my pom (though I honestly don't know why). Removing this explicit version fixed my issue and I can now connect (and more importantly my sanity)
